### PR TITLE
Use numpy.logical_xor to subtract binary images in ImageMath

### LINKED ISSUE #3470: Subtracting two binary masks in ImageMath is broken after numpy 1.14 update
I don't think this is covered in CP's tests, but it IS in a tutorial pipeline; that pipeline built (though the overall repo didn't for a reason I've since fixed) on [January 6th](https://travis-ci.org/CellProfiler/tutorials/builds/325875035#L672) on numpy 1.13.3 but failed on [January 7th](https://travis-ci.org/CellProfiler/tutorials/builds/326139269#L685) after update to numpy 1.14.  Error below.  Pipeline and images at the tutorial repo [here](https://github.com/CellProfiler/tutorials/tree/master/3d_monolayer).

```
Error detected during run of module ImageMath
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/src/cellprofiler/cellprofiler/pipeline.py", line 1770, in run_with_yield
    self.run_module(module, workspace)
  File "/home/travis/virtualenv/python2.7.14/src/cellprofiler/cellprofiler/pipeline.py", line 2022, in run_module
    module.run(workspace)
  File "/home/travis/virtualenv/python2.7.14/src/cellprofiler/cellprofiler/modules/imagemath.py", line 397, in run
    output_pixel_data = op(output_pixel_data, pd)
TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
```

### PR DESCRIPTION
**Pull request by mcquin**
Resolves #3470

### CODE REVIEW

**Comment by bethac07** [comment]
Thanks @mcquin

File: cellprofiler/modules/imagemath.py [hunk 1]
```diff
@@ -505,10 +515,9 @@ def validate_module(self, pipeline):
 
     def upgrade_settings(self, setting_values, variable_revision_number,
                          module_name, from_matlab):
-        if (from_matlab and module_name == 'Subtract' and
-                    variable_revision_number == 3):
+        if (from_matlab and module_name == 'Subtract' and variable_revision_number == 3):
             subtract_image_name, basic_image_name, resulting_image_name, \
-            multiply_factor_1, multiply_factor_2, truncate = setting_values
+                multiply_factor_1, multiply_factor_2, truncate = setting_values
```
Comment (AetherUnbound) [inline cellprofiler/modules/imagemath.py]: Why did this tab get thrown in here?

File: cellprofiler/modules/imagemath.py [hunk 2]
```diff
@@ -573,7 +578,7 @@ def upgrade_settings(self, setting_values, variable_revision_number,
             try:
                 factors += [float(setting_values[2]) *
                             float(setting_values[5])]
-            except:
+            except ValueError:
```
Comment (AetherUnbound) [inline cellprofiler/modules/imagemath.py]: Nice :+1:

File: cellprofiler/modules/imagemath.py (hunk 1 above)
Comment (mcquin) [reply cellprofiler/modules/imagemath.py]: It's the required indentation level for a continuing line (note the `\` ending the line above).

Comment (AetherUnbound) [reply cellprofiler/modules/imagemath.py]: AH okay, the file diff makes it look a little wonky.

**Pull request merged**